dask_ml/model_selection/_incremental.py

```diff
@@ -837,9 +834,44 @@ def _get_params(self):
         if self.n_initial_parameters == "grid":
             return ParameterGrid(self.parameters)
         else:
-            return ParameterSampler(self.parameters, self.n_initial_parameters)
+            return ParameterSampler(
+                self.parameters,
+                self.n_initial_parameters,
+                random_state=self.random_state,
+            )
 
     def _additional_calls(self, info):
+        calls = {k: v[-1]["partial_fit_calls"] for k, v in info.items()}
+        if all(c == 1 for c in calls.values()):
+            # this condition happens at the beginning
+            #
+            # _to_reach records the number of partial_fit calls that need to be
+            # be made on each model to reach the number _adapt specifies.
+            # It only matters for when patience is set -- otherwise,
+            # the calls specified by _adapt are scheduled.
+            #
+            # It's a class variable because this variable needs to be stored
```

Comment:
Sorry, I still don't see this. AFAICT, `_to_reach` isn't used anywhere outside this function (in particular, `_adapt` doesn't use it).

Reply:
> `_adapt` doesn't use it

I implemented it this way so stop-on-plateau can be layered on top of any existing adaptive algorithm with minimal developer work (i.e., so `_adapt` doesn't have to record plateau information). `_adapt` is only called when required by the underlying adaptive algorithm (i.e. Hyperband or successive halving).

`_to_reach` caches the value `_adapt` returns so plateaus can be evaluated more frequently than the number of `partial_fit` calls `_adapt` specifies. If `_adapt` specifies 10 `partial_fit` calls and `patience=3`, the estimator will need to be evaluated every `partial_fit` call to see if it's on a plateau, but `_adapt` should be not called until `partial_fit` has been called 10 times.

Reply:
Ah. I see where you're coming from.

[works a bit]

I now store all the state in `info`. This required some modification of `info`; I have to store the number of calls to `partial_fit` that the adaptive algorithm requires. I can't calculate it without that (which had been the main blocker earlier).

This is a much cleaner implementation – thanks for standing firm.